skcosmo/preprocessing/FlexibleScaler.py

```diff
@@ -0,0 +1,222 @@
+import numpy as np
+import sklearn as sk
+from sklearn.base import TransformerMixin, BaseEstimator
+
+
+class StandardFlexibleScaler(TransformerMixin, BaseEstimator):
+    """Standardize features by removing the mean and scaling to unit variance.
+    Make the mean of the columns equal to zero and the
+    variance of each column (`column_wise==True`) equal to one.
+
+    :param with_mean: If True, center the data before scaling. If False, keep the mean intact
+    :type with_mean: boolean
+    :param with_std: If True, scale the data to unit variance. If False, keep the variance intact
+    :type with_std: boolean
+    :param column_wise: If True, normalize each column separately. If False, normalize the whole matrix, divided it by variaton.
+    :type column_wise: boolean
+    :param tol: The tolerance for the optimization: if the variance are smaller than tol, it is considered zero.
+    """
+
+    def __init__(self, with_mean=True, with_std=True, column_wise=False, tol=1e-15):
+        """Initialize StandardFlexibleScaler."""
+
+        self.with_mean = with_mean
+        self.with_std = with_std
+        self.column_wise = column_wise
+        self.n_samples_seen_ = 0
+        self.tol = tol
+
+    def fit(self, X, y=None):
+        """Compute mean and scaling to be applied for subsequent normalization.
+
+        :param X: Matrix
+        :type X: ndarray
+        :param y: ignored
+
+        :return: itself
+        """
+
+        self.n_samples_seen_, self.n_features_ = X.shape
+        if self.with_mean:
+            self.mean_ = X.mean(axis=0)
+        else:
+            self.mean_ = np.zeros(self.n_features_)
+
+        self.scale_ = 1.0
+        if self.with_std:
+            var = ((X - X.mean(axis=0)) ** 2).mean(axis=0)
+
+            if self.column_wise:
+                if np.any(var < self.tol):
+                    raise ValueError("Cannot normalize a feature with zero variance")
+                self.scale_ = np.sqrt(var)
+            else:
+                var_sum = var.sum()
+                if var_sum < self.tol:
+                    raise ValueError("Cannot normalize a matrix with zero variance")
+                self.scale_ = np.sqrt(var_sum)
+
+        return self
+
+    def transform(self, X, y=None):
+        """Normalize a vector based on previously computed mean and scaling.
+
+        :param X: Matrix
+        :type X: ndarray
+        :param y: ignored
+
+        :return: transformed matrix X
+        """
+
+        if self.n_samples_seen_ == 0:
+            raise sk.exceptions.NotFittedError(
+                "This "
+                + type(self).__name__
+                + " instance is not fitted yet. Call 'fit' with appropriate arguments before using this estimator."
+            )
+        if (self.n_samples_seen_ != X.shape[0]) or (self.n_features_ != X.shape[1]):
```

Comment:
Hey -- sorry I didn't see this before. The X should not necessarily need to match the number of samples, just the number of features. Same with line 108
```suggestion
        if self.n_features_ != X.shape[1]:
```